scootapi/server/serve.go

```diff
@@ -20,16 +24,45 @@ func MakeServer(handler scoot.CloudScoot,
 }
 
 type Handler struct {
-	scheduler scheduler.Scheduler
-	sagaCoord saga.SagaCoordinator
-	stat      stats.StatsReceiver
+	scheduler          scheduler.Scheduler
+	sagaCoord          saga.SagaCoordinator
+	stat               stats.StatsReceiver
+	runJobStatCount    int64
+	jobStatusStatCount int64
+}
+
+func (h *Handler) loop() {
+	tickerS := time.NewTicker(time.Second)
+	tickerM := time.NewTicker(time.Minute)
+	for {
+		// Manually latch and reset until we find a better way to do req/sec within viz's one minute window.
+		// Takes max rps for every one minute window.
+		select {
+		case <-tickerS.C:
+			prevRunRps := h.stat.Gauge("runJobRpsGauge").Value()
+			prevStatusRps := h.stat.Gauge("jobStatusRpsGauge").Value()
+			if prevRunRps < h.runJobStatCount {
+				h.stat.Gauge("runJobRpsGauge").Update(h.runJobStatCount)
+			}
+			if prevStatusRps < h.jobStatusStatCount {
+				h.stat.Gauge("jobStatusRpsGauge").Update(h.jobStatusStatCount)
+			}
+		case <-tickerM.C:
+			h.stat.Gauge("runJobRpsGauge").Update(0)
+			h.runJobStatCount = 0
+			h.jobStatusStatCount = 0
+		}
+	}
 }
 
 func (h *Handler) RunJob(def *scoot.JobDefinition) (*scoot.JobId, error) {
 	defer h.stat.Latency("runJobLatency_ms").Time().Stop()
-	return runJob(h.scheduler, def)
+	h.runJobStatCount++
```

Comment:
is this thread safe? you potentially have concurrent reads & writes of runJobStatCount in this method and in the loop?

Does this pass go test -race ./...